### DISCUSSION TITLE
Can't make Repository’s Secret as Environment variable

### DISCUSSION THREAD

**Discussion by kopach** [renovatebot/renovate#32871] [Request Help] [answered]
### How are you running Renovate?

A Mend.io-hosted app

### Please tell us more about your question or problem

I have admin rights in the following open source repo https://github.com/brainly/gene
In Renovate dashboard (https://developer.mend.io/github/brainly/gene/-/settings) I've added new Repository secret `MEND_GITHUB_REGISTRY_TOKEN`
I would like to make this as `Env var`. But, when I click the checkbox - I'm getting UI error  "Error updating secret MEND_GITHUB_REGISTRY_TOKEN". In Network, I can see response `User does not have permission to access`. 
What permissions am I missing here? I'm an admin (most advanced role for the repo possible)

Could you help me here, please?

### Logs (if relevant)

![2024-12-03 12 40 37](https://github.com/user-attachments/assets/fa4d75b8-af09-43c0-9c28-5d52f3cd039d)

![image](https://github.com/user-attachments/assets/2c67eacc-2f1a-43b6-a991-399f7f4b95e0)

**Comment by kopach** [comment]
I believe this could be related https://github.com/renovatebot/renovate/discussions/32205

**Comment by rarkins** [comment]
Our backend team will look into it

**Reply by kopach** [reply]
Thanks, @rarkins. Do you know any workarounds to this, maybe in the meanwhile? Basically, I would like renovate to have access to private npm registry hosted on GH using GH PAT + host rule. Some day ago it was possible to [encrypt secret](https://app.renovatebot.com/encrypt), but this doesn't work any ore

**Reply by rarkins** [reply]
It was never possible to configure env vars before. As far as I'm aware, you can do just as much as you used to. The secrets you configure (without env var box ticked) should be passed through in jobs as `secrets`.

**Reply by kopach** [reply]
Thanks @rarkins for your reply. I can confirm now that `env var` box ticket is not required in this case.
I finally got it working, however **documentation is misleading/incorrect**
here https://docs.renovatebot.com/mend-hosted/credentials/#adding-a-secret it's stated, that 
```json
{
  "hostRules": [
    {
      "matchHost": "github.com",
      "token": "{{ secrets.MY_ORG_SECRET }}"
    }
  ]
}
```
will be enough. However, to make it really work
1. I have to add `hostType: 'npm',`
2. I have to update `matchHost` to something like this `matchHost: 'npm.pkg.github.com',` just ` "matchHost": "github.com"` doesn't work

So, finally I got it working with config like so in my `.github/renovate.json5`

```json5
    {
      hostType: 'npm',
      matchHost: 'npm.pkg.github.com',
      token: '{{ secrets.MEND_GITHUB_REGISTRY_TOKEN }}',
    },
```

**Reply by kopach** [reply]
However, permission issue with `env var` still remains (even though not critical for me anymore). 
I also can't remove rules I created in renovate dashboard myself, as well as can't remove secrets I've added myself ¯\_(ツ)_/¯
I would like to clean up those. Rules issue is described here also https://github.com/renovatebot/renovate/discussions/32205

**Reply by rarkins** [reply, marked as answer]
>I finally got it working, however **documentation is misleading/incorrect** here [docs.renovatebot.com/mend-hosted/credentials#adding-a-secret](https://docs.renovatebot.com/mend-hosted/credentials/#adding-a-secret) it's stated, that
> 
> ```json
> {
>   "hostRules": [
>     {
>       "matchHost": "github.com",
>       "token": "{{ secrets.MY_ORG_SECRET }}"
>     }
>   ]
> }
> ```
> 
> will be enough. 

It doesn't state that last part at all. The docs there show how to reference a UI secret in hostRules, and does not say "this is the exhaustive documentation for hostRules" or "this is all you need to do".

If you have concrete suggestions for adding/improving, PRs for docs are welcome in this repo
